# HG changeset patch
# User Rik <rik@octave.org>
# Date 1355182718 28800
#      Mon Dec 10 15:38:38 2012 -0800
# Node ID 0a27a0e157a6340e1f1ea7a5fb00c93fc321d870
# Parent  56239ff815a34fecc1ff5522b5affd2174b86ed9
Temporary fix for display of indexed images of integer class.
True fix will be done later in C++ code.

* image.m: Convert integer indexed images to class single and
switch to 1-based indexing.

diff --git a/scripts/image/image.m b/scripts/image/image.m
--- a/scripts/image/image.m
+++ b/scripts/image/image.m
@@ -100,16 +100,23 @@ endfunction
 function h = __img__ (x, y, img, varargin)
   
   newplot ();
 
   if (isempty (img))
     error ("__img__: matrix is empty");
   endif
 
+  ## FIXME: Hack for integer formats which use zero-based indexing
+  ##        Hack favors correctness of display over size of image in memory.
+  ##        True fix will be done in C++ code. 
+  if (ndims (img) == 2 && (isinteger (img) || islogical (img)))
+    img = single (img) + 1;
+  endif
+
   if (isempty (x))
     x = [1, columns(img)];
   endif
 
   if (isempty (y))
     y = [1, rows(img)];
   endif
 
